test/powershell/Host/Startup.Tests.ps1

```diff
@@ -0,0 +1,120 @@
+            'System.Net.Primitives.dll'
+            'System.ObjectModel.dll'
+            'System.Private.CoreLib.dll'
+            'System.Private.Uri.dll'
+            'System.Private.Xml.dll'
+            'System.Private.Xml.Linq.dll'
+            'System.Reflection.dll'
+            'System.Reflection.Emit.dll'
+            'System.Reflection.Emit.ILGeneration.dll'
+            'System.Reflection.Emit.Lightweight.dll'
+            'System.Reflection.Extensions.dll'
+            'System.Reflection.Primitives.dll'
+            'System.Resources.ResourceManager.dll'
+            'System.Runtime.dll'
+            'System.Runtime.Extensions.dll'
+            'System.Runtime.InteropServices.dll'
+            'System.Runtime.InteropServices.RuntimeInformation.dll'
+            'System.Runtime.Loader.dll'
+            'System.Runtime.Numerics.dll'
+            'System.Runtime.Serialization.Formatters.dll'
+            'System.Runtime.Serialization.Primitives.dll'
+            'System.Security.AccessControl.dll'
+            'System.Security.Claims.dll'
+            'System.Security.Cryptography.Algorithms.dll'
+            'System.Security.Cryptography.Encoding.dll'
+            'System.Security.Cryptography.Primitives.dll'
+            'System.Security.Cryptography.X509Certificates.dll'
+            'System.Security.Principal.dll'
+            'System.Security.Principal.Windows.dll'
+            'System.Text.Encoding.CodePages.dll'
+            'System.Text.Encoding.Extensions.dll'
+            'System.Text.RegularExpressions.dll'
+            'System.Threading.dll'
+            'System.Threading.AccessControl.dll'
+            'System.Threading.Tasks.dll'
+            'System.Threading.Tasks.Parallel.dll'
+            'System.Threading.Thread.dll'
+            'System.Threading.ThreadPool.dll'
+            'System.Threading.Timer.dll'
+            'System.Xml.ReaderWriter.dll'
+            'System.Xml.XDocument.dll'
+        )
+
+        if ($IsWindows) {
+            $expectedAssemblies += @(
+                'Microsoft.Management.Infrastructure.CimCmdlets.dll'
+                'Microsoft.PowerShell.CoreCLR.Eventing.dll'
+                'System.DirectoryServices.dll'
+                'System.Management.dll'
+                'System.Reflection.Metadata.dll'
+                'System.Security.Permissions.dll'
+                'System.Threading.Overlapped.dll'
+            )
+        }
+        else {
+            $expectedAssemblies += @(
+                'System.IO.MemoryMappedFiles.dll'
+                'System.Net.Sockets.dll'
+                'System.Reflection.Metadata.dll'            )
+        }
+
+        $loadedAssemblies = pwsh -noprofile -command '([System.AppDomain]::CurrentDomain.GetAssemblies()).manifestmodule | ? { $_.Name -notlike ""<*>"" } | % { $_.Name }'
+    }
+
+    It "No new assemblies are loaded" {
+        foreach ($assembly in $loadedAssemblies) {
+            $expectedAssemblies | Should -Contain $assembly
```

Comment:
Naming is a little confusing.
It feels like `$expectedAssemblies` means `when process starts, I expect these assemblies to be there`.
However currently condition won't catch a situation where some/several `expectedAssemblies` is Not loaded.
Probably change the name from `$expectedAssemblies` to something like `$allowedAssemblies` (or make the list comparison more strict).